### DISCUSSION TITLE
Kaspersky integration

### DISCUSSION THREAD

**Discussion by asrachid** [Security-Onion-Solutions/securityonion#12530] [Unsupported Versions]
Hello! I want to integrate kaspersky to security onion.
My case : KSC version is 14.2, SO version is 2.4
I use a distributed architecture: a manager, a fleet node, a search node and a forward node.

**Comment by cm-ops** [comment]
Have you looked at this? https://support.kaspersky.com/datafeeds/integration/15474

**Reply by asrachid** [reply]
I have already seen this tutorial, but in my case I have Kaspersky Center, I don't want to receive data feeds but I'm interested in Kaspersky logs like viruses detected etc.

**Reply by asrachid** [reply]
I want to know how to parse Kaspersky logs in syslog format

**Reply by cm-ops** [reply]
You can certainly send them via syslog https://docs.securityonion.net/en/2.4/syslog.html#syslog, they will be parsed as syslog though. If you wanted them parsed out differently you would need to create a custom parser for the logs.

**Reply by asrachid** [reply]
I create a custom parser and I put it in "/opt/so/saltstack/local/salt/elasticsearch/files/ingest/", now I want to know where I should specify which logs should be parsed using this custom parser.

**Reply by cm-ops** [reply]
You would also need to create a custom Logstash pipeline to bring the logs in and send to Elasticsearch for parse (with your custom parser) and ingest. 

If you are able to put an Elastic Agent on the server, you could set up a custom log integration policy.

**Reply by asrachid** [reply]
Logstash will just forward the logs to elasticsearch, but where in elasticsearch should I specify which parser to use for the particular logs we are sending.

**Reply by cm-ops** [reply]
In your Logstash output to Elasticsearch, you would specify the pipeline to send it to. https://www.elastic.co/guide/en/logstash/current/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-pipeline
